protocol.md

```diff
@@ -59,39 +59,41 @@ If no [Local Encrypted Storage] is found, *Tanker Core* will create one automati
 
-First, if they are not found, *Tanker Core* creates a [Device Encryption Key Pair], [Device Signature Key Pair], for the *physical device*, and a [User Encryption Key Pair]. It stores them in the [Local Encrypted Storage].
+*Tanker Core* creates a [User Encryption Key Pair].
 
-*Tanker Core* creates another [Device Encryption Key Pair] and [Device Signature Key Pair] for the *virtual device*. But instead of storing them, it serializes them as an opaque token, the [Verification Key].
+*Tanker Core* creates a [Device Encryption Key Pair] and a [Device Signature Key Pair] for the *virtual device*. It serializes them as an opaque token, the [Verification Key].
 
-A first `device_creation` block is constructed with the *virtual device*'s public [Device Encryption Key Pair] and public [Device Signature Key Pair] and the encrypted [User Encryption Key Pair]. This block is signed with the ephemeral private key of the [Delegation Token]found in the [Secret Permanent Identity].
+A first `device_creation` block is constructed with the *virtual device*'s public [Device Encryption Key Pair] and public [Device Signature Key Pair] and the encrypted [User Encryption Key Pair]. This block is signed with the ephemeral private key of the [Delegation Token] found in the [Secret Permanent Identity].
 
-Then, a second `device_creation` block is created with the *physical device*'s [Device Encryption Key Pair] and [Device Signature Key Pair]. *Tanker Core* creates a new ephemeral key pair, signs this block with the ephemeral private key and uses the *virtual device*'s [Device Signature Key Pair] to sign the *delegation* and fill the block's *delegation_signature* field.
+*Tanker Core* creates a [Device Encryption Key Pair] and a [Device Signature Key Pair] for the *physical device*.
+
+A second `device_creation` block is created with the *physical device*'s [Device Encryption Key Pair] and [Device Signature Key Pair]. *Tanker Core* creates a new ephemeral key pair, signs this block with the ephemeral private key and uses the *virtual device*'s [Device Signature Key Pair] to sign the *delegation* and fill the block's *delegation_signature* field.
 
 Finally, both `device_creation` *block*s are pushed to the *Trustchain*, the *virtual device* before the *physical one*, and the [Verification Key] is uploaded to the *Tanker server* according to the [Verification Method] used by the *user*.
 
-Assuming the pushed *block*s are correct, the *Tanker server* acknowledges them, allowing the just created *device* to proceed and [authenticate itself](#device-authentication).
+Assuming the pushed *block*s are correct, the *Tanker server* stores them, and initiates an authenticated session for the *physical device*. The just created *physical device* permanently stores its [Device Encryption Key Pair] and [Device Signature Key Pair] in the [Local Encrypted Storage].
```

Comment:
Aren't they stored before the blocks are sent to the server?

Reply:
Not anymore!

Reply:
Hmm I see ^^